### ISSUE TITLE
Add full support for Roborock S7 MaxV Ultra

### ISSUE THREAD

**Issue by golddragon007** [rytilahti/python-miio#1415]
**Is your feature request related to a problem? Please describe.**
The maxv ultra has some maintenance and other features which is not supported right now by the integration.
These are like:
- mopping related features (like intensity, wash settings)
- station related features (like bag, auto empty, water level warning)
- maintenance (like the water filter, high-speed maintenance brush, dustbag)
- reactive AI
- different cleanup modes (like vacuum & mop, vacuum, mop, custom)
- firmware update (see if there is new and trigger it)

**Describe the solution you'd like**
Possibly to have the options to select these on the UI.

**Device information:**
If the enhancement is device-specific, please include also the following information.

  - Name(s) of the device: Roborock S7 MaxV Ultra
  - Link: https://us.roborock.com/pages/roborock-s7-maxv

Use `miiocli device --ip <ip address> --token <token>`.

  - Model: roborock.vacuum.a27
  - Hardware version: 
  - Firmware version: 4.1.5_4236

**Additional context**

**Comment by DarkByteZero** [comment]
Why wrong project @golddragon007 ?  miio/integrations/vacuum/roborock/vacuumcontainers.py for example declares the different maintanance metrics.

**Comment by golddragon007** [comment]
The wrong project, because I wanted to comment on the integration of Home assistants. As I don't know which features are not implemented by whom. If you see that some of my mentions are not implemented by the library, I can reopen this ticket.

**Comment by rytilahti** [comment]
Just to give some background, this library was initially created mainly to add support to the roborock vacuum (gen1) but got then extend thanks to all contributors. At that time, it was really simple to hardcode the available sensors in homeassistant to make it all work nicely, but this does not really scale as support for new devices with different sensors were introduced.

The real fix would be extending this library to make it possible to provide information about available sensors through this library (some ideas are written down in #1114), but, alas, I haven't had time to implement that and that is unlikely to change in the near future. So if someone is interested in pursuing adding introspectable sensors, I'm happy to help with the effort by nudging to correct direction. :-)

**Comment by DarkByteZero** [comment]
Yeah, then this issue should still be opened again because S7 MAXV Ultra is not fully supported by this library. Maybe I can help implement some changes.

**Comment by BillyFKidney** [comment]
> * different cleanup modes (like vacuum & mop, vacuum, mop, custom)

This will be a huge improvement in the integration.
When using vacuum only the S7 MaxV doesn't wash the mop first, reducing water usage.
